### DISCUSSION TITLE
v0.7.0-2

### DISCUSSION THREAD

**Discussion by ailisp** [near/near-sdk-js#330] [Announcements]
## Highlights
- Docstrings are now available on every public APIs.
- Support of WSL 2. You can now develop with near-sdk-js on Windows! Check README for recipes.
- ABI support is added on CLI!
- CLI enhancement: flexible build commands are added!
- Starting from this version, near-contract-standards is released together with near-sdk-js!
- UTF-8 support is added. You can now store and retrieve UTF-8 strings in storage and collections.
- The confusing Bytes types is dropped. It is replaced with string and Uint8Array.
- Storage/Input/ValueReturn API now comes with both raw version and string version. The raw version will help you efficiently handle raw bytes in Uint8Array.
- Refactor: starting from this version, near-sdk-js becomes a monorepo with a number of enhancement for contributors.
- Refactor: Optimized exports.

## What's Changed
* v0.6.0 by @volovyks in https://github.com/near/near-sdk-js/pull/275
* deps dir management moved to post-install script by @volovyks in https://github.com/near/near-sdk-js/pull/276
* TSC error suppression for latest versions by @vunderkind in https://github.com/near/near-sdk-js/pull/272
* cli chmod hack deleted by @volovyks in https://github.com/near/near-sdk-js/pull/277
* patch(cli): fix typo by @waynenilsen in https://github.com/near/near-sdk-js/pull/282
* NFT standard by @ailisp in https://github.com/near/near-sdk-js/pull/177
* cli bash replaced with node by @volovyks in https://github.com/near/near-sdk-js/pull/285
* quick js v0.1.3 by @volovyks in https://github.com/near/near-sdk-js/pull/279
* ts-morph to deps by @idea404 in https://github.com/near/near-sdk-js/pull/292
* standarts import paths fixed by @volovyks in https://github.com/near/near-sdk-js/pull/293
* Test cross contract call failures and error handling by @ailisp in https://github.com/near/near-sdk-js/pull/294
* WSL2 instructions by @volovyks in https://github.com/near/near-sdk-js/pull/295
* Added a warning until we can get the runtime audited by @DavidM-D in https://github.com/near/near-sdk-js/pull/298
* Exports and import paths by @volovyks in https://github.com/near/near-sdk-js/pull/299
* feat(cli): new subcommands by @waynenilsen in https://github.com/near/near-sdk-js/pull/283
* NFT standard refund account storage based on actual size by @ailisp in https://github.com/near/near-sdk-js/pull/300
* [bugfix] display stderr on command failure by @DerekTBrown in https://github.com/near/near-sdk-js/pull/303
* Example of programmatic update of JS Smart Contract by @nsejim in https://github.com/near/near-sdk-js/pull/287
* add remaining doc strings by @ailisp in https://github.com/near/near-sdk-js/pull/305
* Include Example and Test for Promises Created in For Loops by @idea404 in https://github.com/near/near-sdk-js/pull/309
* fix: Fix logging by @petarvujovic98 in https://github.com/near/near-sdk-js/pull/311
* Use TS ? to replace option in nft standard interface by @ailisp in https://github.com/near/near-sdk-js/pull/304
* Exports refactored by @volovyks in https://github.com/near/near-sdk-js/pull/313
* ABI generation by @itegulov in https://github.com/near/near-sdk-js/pull/301
* pnpm by @volovyks in https://github.com/near/near-sdk-js/pull/317
* fix: ensure that NEAR functions have explicit types by @itegulov in https://github.com/near/near-sdk-js/pull/322
* Separate tests replased with one script in examples CI by @volovyks in https://github.com/near/near-sdk-js/pull/318
* Monorepo implemented with Turborepo by @volovyks in https://github.com/near/near-sdk-js/pull/323
* Use uint8array by @ailisp in https://github.com/near/near-sdk-js/pull/308
* move IntoStorageKey alike feature to lib by @ailisp in https://github.com/near/near-sdk-js/pull/329

## New Contributors
* @vunderkind made their first contribution in https://github.com/near/near-sdk-js/pull/272
* @waynenilsen made their first contribution in https://github.com/near/near-sdk-js/pull/282
* @DavidM-D made their first contribution in https://github.com/near/near-sdk-js/pull/298
* @DerekTBrown made their first contribution in https://github.com/near/near-sdk-js/pull/303
* @nsejim made their first contribution in https://github.com/near/near-sdk-js/pull/287

**Full Changelog**: https://github.com/near/near-sdk-js/compare/v0.6.0...v0.7.0-2

<hr /><em>This discussion was created from the release <a href='https://github.com/near/near-sdk-js/releases/tag/v0.7.0-2'>v0.7.0-2</a>.</em>

**Comment by ailisp** [comment]
To test it in your project, simply use  `"near-sdk-js": "^0.7.0-2",` and optionally `"near-contract-standards": "^0.7.0-2",` in your project's package.json. So far it works well for me.

**Comment by idea404** [comment]
Exciting stuff! Some additions here merit a little explanation also in [near.docs](https://docs.near.org/) such as the WSL support and CLI use. Something DevRel can pick up in the next few weeks :)

**Reply by ailisp** [reply]
Thank you Dennis! I've created an issue to track what you said: https://github.com/near/docs/issues/1326
